src/components/common/Main/cmp.tsx

```diff
@@ -3,14 +3,9 @@ import { MainProps } from './types'
 import { useConnect } from '@/hooks/common/useConnect'
 
 export const Main = ({ children }: MainProps) => {
-  const { isConnected, connect } = useConnect()
+  const { isConnected, connect, tryReconnect } = useConnect()
 
   useEffect(() => {
-    async function tryReconnect() {
-      if (isConnected) return
-      await connect()
-    }
-
     tryReconnect()
   }, [connect, isConnected])
```

Comment:
update the list of deps here with just `tryReconnect`. Your IDE should warn you about this, isn't it?